docx_editor/session.py

```
@@ -131,29 +146,111 @@ def _read_pid(connection_file: Path) -> int | None:
+    """start_session's budget: DOCX_SESSION_START_TIMEOUT, else the 30 s default.
+
+    Raises:
+        SessionError: If the variable is set to anything but a positive number.
+    """
+    raw = os.environ.get(START_TIMEOUT_ENV, "").strip()
+    if not raw:
+        return DEFAULT_START_TIMEOUT
+    try:
+        value = float(raw)
+        if not value > 0:  # also rejects nan
+            raise ValueError
```

Ruff (0.16.2)

[warning] Abstract `raise` to an inner function

(TRY301)